app/notification_test.go

```diff
@@ -817,3 +819,287 @@ func TestDoesStatusAllowPushNotification(t *testing.T) {
 		t.Fatal("Should have been false")
 	}
 }
+
+func TestGetDirectMessageNotificationEmailSubject(t *testing.T) {
+	Setup()
+	expectedPrefix := "[http://localhost:8065] New Direct Message from sender on"
+	post := &model.Post{
+		CreateAt: 1501804801000,
+	}
+	translateFunc := utils.GetUserTranslations("en")
+	subject := getDirectMessageNotificationEmailSubject(post, translateFunc, "http://localhost:8065", "sender")
+	if !strings.HasPrefix(subject, expectedPrefix) {
+		t.Fatal("Expected subject line prefix '" + expectedPrefix + "', got " + subject)
+	}
+}
+
+func TestGetNotificationEmailSubject(t *testing.T) {
+	Setup()
+	expectedPrefix := "[http://localhost:8065] Notification in team on"
+	post := &model.Post{
+		CreateAt: 1501804801000,
+	}
+	translateFunc := utils.GetUserTranslations("en")
+	subject := getNotificationEmailSubject(post, translateFunc, "http://localhost:8065", "team")
+	if !strings.HasPrefix(subject, expectedPrefix) {
+		t.Fatal("Expected subject line prefix '" + expectedPrefix + "', got " + subject)
+	}
+}
+
```

Comment:
If you wanted, you could make this a two step process like
1) Have `getNotificationEmailBody` return a template with props set
2) Have the caller call render on it
so that you can check the individual props without worrying about the actual text of the message. 0/5 on that though